enhancements/network/ovn-multi-external-gateway.md

```diff
@@ -0,0 +1,182 @@
+| 4.13                  | Yes                   | Yes           | Use annotations to configure external gateways    |
+| 4.13                  | Yes                   | No            | Use annotations to configure external gateway     |
+| 4.13                  | No                    | No            | Nothing to do                                     |
+| 4.14                  | No                    | Yes           | Use CR information to configure external gateways |
+| 4.14                  | Yes                   | Yes           | Use CR information to configure external gateways |
+| 4.14                  | Yes                   | No            | Use annotations to configure external gateways    |
+| 4.14                  | No                    | No            | Nothing to do                                     |
+
+The current annotation based logic will be enhanced as part of this change to support the logic described in the previous table.
+
+### Workflow Description
+The workflow follows the behavior of a namespaced CR.
+
+
+### API Extensions
+
+A new namespaced-scoped CRD is created to represent the multiple external gateways configuration. The `spec` structure exposes the following fields:
+* `namespaces` contains a slice of the namespaces linked to this gateway configuration. It maps to the `k8s.ovn.org/routing-namespaces` annotation.
+* `staticGateways` allows defining the configuration for external gateways. It exposes the following fields:
+  * `externalIPs`: contains a slice of IPv4 or IPv6 addresses that are configured as static external gateways. It maps to the `k8s.ovn.org/routing-external-gws` annotation.
+  * `bfdEnabled` determines if Bi-Directional Forwarding Detection (BFD) is supported for the static gateway IPs defined in the `staticGateways` field. It's enabled when set to true and disabled when false or omited in the manifest. It maps to the `k8s.ovn.org/bfd-enabled` annotation in the namespace.
+* `podGateways` allows defining the selector and configuration criteria for the gateway pods. This configuration contains the following fields:
+  * `network` references the multus network configuration that matches the same name in the pod that will be used for external traffic. If the field is empty or undefined, the pod's IP will be used instead. This matches the current behavior used with annotations. This field maps to the annotation `k8s.ovn.org/routing-network`.
+  * `bfdEnabled` determines if Bi-Directional Forwarding Detection (BFD) is supported in this network and follows the same behavior as with the annotation: it's enabled when set to true and disabled when false or omited in the manifest. It maps to the `k8s.ovn.org/bfd-enabled` annotation in the pod.
+  * `podSelector` implements a [set-based](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#set-based-requirement) label selector to filter the pods in the namespace that match this network configuration.
```

Comment:
I am not following the purpose of this new `podSelector` field and what is this translating to from the annotations world, help needed in understanding and sorry for being dumb.

Reply:
Sure, no problem :smile:  : The current implementation uses pod's annotations to identify the pods that perform the role of gateways. Moving to a CRD, these annotations are no longer expected to be found in the pod's manifest. Instead, we have to resort using filtering criteria to identify the pods that are meant to be gateways. This label selector is exactly that: defines the filtering criteria via label matching that identifies the running pods to be processed as gateways.

Reply:
ah thanks for explaining! makes sense...